### PR TITLE
Django command for checking Celery queue fullness

### PR DESCRIPTION
**Pull request by radimsuckr**

### CODE REVIEW

File: security/config.py [hunk 1]
```diff
@@ -44,6 +44,8 @@
     'BACKUP_STORAGE_PATH': None,
     'DEBUG_TOOLBAR': False,
     'SHOW_DEBUG_TOOLBAR': False,
+    'CELERY_HEALTH_CHECK_DEFAULT_QUEUE': 'default',
+    'CELERY_HEALTH_CHECK_MAX_TASKS_COUNT': 50,
```
Comment (radimsuckr) [inline security/config.py]: Je otázka, co je tady "sane default".

File: security/management/commands/celery_health_check.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    help = 'Check number of waiting Celery tasks in given queue'
+    messages = {
+        'error': 'There are more than {} waiting tasks in queue "{}" and therefore it is considered as full.',
+        'invalid_max_tasks_count': 'Max tasks count must be >= 0.',
+        'ok': 'Queue "{}" is ok.',
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-q', '--queue_name',
```
Comment (radimsuckr) [inline security/management/commands/celery_health_check.py]: Underscore to dash.

Comment (rubickcz) [inline security/management/commands/celery_health_check.py]:
```suggestion
            '-q', '--queue-name',
```

Comment (radimsuckr) [inline security/management/commands/celery_health_check.py]:
```suggestion
            '-q', '--queue-name',
```

File: security/management/commands/celery_health_check.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            type=int,
+        )
+
+    def handle(self, *args, **options):
+        max_tasks_count = options['max_tasks_count']
+        if max_tasks_count < 0:
+            raise CommandError(self.messages['invalid_max_tasks_count'])
+        queue_name = options['queue_name']
+
+        tasks = CeleryTaskLog.objects.filter(is_set_as_stale=False, queue_name=queue_name).exclude(
```
Comment (rubickcz) [inline security/management/commands/celery_health_check.py]: Why aren't you simply filtering CeleryTaskLogs with state `WAITING`?

File: security/config.py (hunk 1 above)
Comment (rubickcz) [reply security/config.py]: IMO highly depends on project... but this looks ok to me.

**Comment by rubickcz** [comment]
Tests failed.

File: security/management/commands/celery_health_check.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            ).values('celery_task_id'),
+        ).order_by('created_at')
+
+    def _handle_max_created_at_diff(self, queryset, queue_name, max_created_at_diff):
+        if max_created_at_diff < 0:
+            raise CommandError(self.messages['invalid_max_created_at_diff'])
+
+        task = queryset.annotate(diff=ExpressionWrapper(Now() - F('created_at'), output_field=DurationField())).filter(
+            diff__gt=timedelta(seconds=max_created_at_diff),
+        ).first()
```
Comment (Formulka) [inline security/management/commands/celery_health_check.py]: exists instead of first

File: security/management/commands/celery_health_check.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+        queue_name = options['queue_name']
+        queryset = self._get_queryset(queue_name)
+
+        if max_created_at_diff:
+            self._handle_max_created_at_diff(queryset, queue_name, max_created_at_diff)
+        if max_tasks_count:
+            self._handle_max_tasks(queryset, queue_name, max_tasks_count)
+
+        self.stdout.write(self.style.SUCCESS(self.messages['ok'].format(queue_name)))
```
Comment (Formulka) [inline security/management/commands/celery_health_check.py]: this seems to get to the output even when an error is raised

File: security/config.py [hunk 6]
```diff
@@ -44,6 +44,7 @@
     'BACKUP_STORAGE_PATH': None,
     'DEBUG_TOOLBAR': False,
     'SHOW_DEBUG_TOOLBAR': False,
+    'CELERY_HEALTH_CHECK_DEFAULT_QUEUE': 'default',
```
Comment (matllubos) [inline security/config.py]: it should get from CELERY_DEFAULT_QUEUE

File: example/apps/test_security/tests/__init__.py [hunk 7]
```diff
@@ -570,3 +579,75 @@ def test_task_log_is_processing_should_return_if_task_is_active_or_waiting(self,
...
+
+        with assert_raises(CommandError):
+            call_command('celery_health_check', max_created_at_diff=-1, stdout=stdout, stderr=stderr)
+            assert_equal(stdout.getvalue(), '')
+            assert_equal(stderr.getvalue(), CeleryHealthCheckCommand.messages['invalid_max_created_at_diff'])
+
+    @freeze_time(now())
+    @data_provider('create_celery_task_log_without_celery_task_run_log')
+    def test_celery_health_check_command_does_not_raise_error_if_queue_is_within_max_created_at_diff_limit(self,
+                                                                                                           task_log):
```
[the remark this answers is not in the document]
Comment (radimsuckr) [reply example/apps/test_security/tests/__init__.py]: Line 628 would be over 120 characters.

**Comment by matllubos** [comment]
rebase please

**Pull request merged**